src/index.js

```diff
@@ -20,8 +20,23 @@ function IpfsAPI (host_or_multiaddr, port) {
     config.host = maddr.address
     config.port = maddr.port
   } catch (e) {
-    config.host = host_or_multiaddr
-    config.port = port || config.port
+    if (typeof host_or_multiaddr === 'string') {
+      config.host = host_or_multiaddr
+      config.port = port && typeof port !== 'object' ? port : config.port
+    }
+  }
+
+  let lastIndex = arguments.length
+  while (!opts && lastIndex-- > 0) {
+    opts = arguments[lastIndex]
+    if (opts) break
+  }
```

Comment:
It's preventing config.host or config.port from being set to the opts object, thus preserving the defaults if needed
Setting host or port to opts is not problematic if you set host or port in opts. If those aren't set the defaults are overridden with the opts object and we can't connect
e.g. so `ipfsAPI({protocol: 'http'})` will work